### PR TITLE
Some fixes for PHPStan level 2 on AudienceTargetingBundle

### PR DESCRIPTION
**Pull request by rogoit**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Easy fixes in AudienceTargetingBundle for PHPStan level 2

#### Why?

Open source is passion and i want to give something back

### CODE REVIEW

File: src/Sulu/Bundle/CategoryBundle/Api/Category.php [hunk 1]
```diff
@@ -460,20 +461,23 @@ private function getUserFullName($user)
     /**
      * Takes an array of CollectionMeta and returns a single meta for a given id.
      *
-     * @param $meta
-     * @param $id
+     * @param CollectionMeta $meta
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Api/Category.php:464]: i think this should be `CategoryMetaInterface[]`

File: src/Sulu/Bundle/CategoryBundle/Api/Category.php [hunk 2]
```diff
@@ -460,20 +461,23 @@ private function getUserFullName($user)
...
+     * @param int $id
      *
-     * @return CollectionMeta
+     * @return null|CollectionMeta
      */
     private function getSingleMetaById($meta, $id)
     {
         if (null !== $id) {
+            /** @var CollectionMeta $singleMeta */
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Api/Category.php:472]: this should not be necessary if we use `CategoryMetaInterface[]` for the $meta param

File: src/Sulu/Bundle/CategoryBundle/Api/Category.php [hunk 3]
```diff
@@ -460,20 +461,23 @@ private function getUserFullName($user)
     /**
      * Takes an array of CollectionMeta and returns a single meta for a given id.
      *
-     * @param $meta
-     * @param $id
+     * @param CollectionMeta $meta
+     * @param int $id
      *
-     * @return CollectionMeta
+     * @return null|CollectionMeta
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Api/Category.php:467]: this should be `CategoryMetaInterface `

File: src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php [hunk 4]
```diff
@@ -282,7 +283,7 @@ public function move($id, $parent)
      * and provides neat getters and setters. If the given object is already an API-object,
      * the associated entity is used for wrapping.
      *
-     * @param $category
+     * @param Category $category
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php:286]: think this should be `CategoryInterface`

File: src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php [hunk 5]
```diff
@@ -303,7 +304,7 @@ public function getApiObject($category, $locale)
      * Returns an array of API-Objects for a given array of category-entities.
      * The returned array can contain null-values, if the given entities are not valid.
      *
-     * @param $entities
+     * @param null|CategoryWrapper $entities
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php:307]: think this should be `CategoryInterface[]`

File: src/Sulu/Bundle/CategoryBundle/Entity/CategoryRepositoryInterface.php [hunk 6]
```diff
@@ -30,7 +30,7 @@ public function isCategoryId($id);
     /**
      * Returns true if the given key is assigned to an existing category.
      *
-     * @param $key
+     * @param string|int $key
```
Comment (nnatter) [inline src/Sulu/Bundle/CategoryBundle/Entity/CategoryRepositoryInterface.php:33]: think this should be `string`

**Review by nnatter — COMMENTED** [review]
Thanks for your effort! Added a few suggestions 🙂

**Comment by nnatter** [comment]
I just tried to rebase this branch to `release/2.1`. Unfortunately, I think something went wrong because I do not have write permission on your fork. Can you give me permissions to push the rebased version of the branch?

**Comment by rogoit** [comment]
You have invite ;)

**Review by nnatter — APPROVED** [review]
Thanks a lot for your contribution! 🙂

**Pull request merged**